### ISSUE TITLE
Parsing failed: Unexpected extern

### ISSUE THREAD

**Issue by nadako** [adireddy/haxe-checkstyle#35]
This one comes from the `import haxe.extern.EitherType;` I believe.

```
Stacktrace: 
Called from haxeparser/HaxeParser.hx line 602
Called from haxeparser/HaxeParser.hx line 513
Called from haxeparser/HaxeParser.hx line 504
Called from haxeparser/HaxeParser.hx line 504
Called from haxeparser/HaxeParser.hx line 495
Called from haxeparser/HaxeParser.hx line 261
Called from Checker.hx line 103
Called from Checker.hx line 122
```

**Comment by AlexHaxe** [comment]
https://github.com/Simn/haxeparser/pull/28

**Comment by AlexHaxe** [comment]
@Simn has already merged my fix, but now we have to get @mcheshkov to pull those changes into haxeparser-substituted

**Comment by nadako** [comment]
What's the deal with the `-substituted` thing? Can't it be merged with haxeparser?

**Comment by Simn** [comment]
I think it's obsolete because the `@:generic` issues in Haxe are fixed now.

**Comment by AlexHaxe** [comment]
I have not really looked into the differences between the two versions, but right now checkstyle uses  the `TypeDecl` typedef from `haxeparser-substituted`. 
If we can get rid of `TypeDecl` we should be able to switch back to haxeparser.

**Comment by adireddy** [comment]
It's not a simple switch but as @Simn mentioned since all the `@:generic` issues are fixed now we should start looking into using `haxeparser` and `hxparse` instead of `haxeparser-substituted`.

**Comment by mcheshkov** [comment]
`haxeparser-substituted` was only to deal with `@:generic` issues and our build system. It should not contain any differences, but substituted by hand type parameters.

**Comment by AlexHaxe** [comment]
I see `TypeDecl` is already in `haxeparser`, the only problem is, it just hasn't been uploaded to haxelib.
So switching should be fairly easy, at least there should be no compile problems.

**Comment by adireddy** [comment]
Thanks @mcheshkov.

@AlexHaxe Yes it works, just tried by manually updating `haxeparser`.

@Simn Any chance of releasing `haxeparser` to haxelib soon?

**Comment by Simn** [comment]
done

**Comment by adireddy** [comment]
Thanks @Simn :+1:

**Comment by adireddy** [comment]
@nadako updated and released new version to haxelib v1.2.2
